samples/HelloSignalR/Providers/AuthenticationProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿using AspNet.Security.OpenIdConnect.Server;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Authentication.OpenIdConnect;
+using AspNet.Security.OpenIdConnect.Extensions;
+
+namespace HelloSignalR.Providers {
+    public class AuthenticationProvider : OpenIdConnectServerProvider {
+        public override Task ValidateClientAuthentication(ValidateClientAuthenticationContext context) {
+            if (context.ClientId == "AspNetContribSample") {
+                context.Validated();
```

Comment:
Actually, good samples are a great start for documentation, because then we can write it more easily by simply explaining some code and referring to working samples.